File: java/space-age/src/main/java/SpaceAge.java
class SpaceAge {
    double _seconds;
    double _onEarth;
    SpaceAge(double seconds) {
        _seconds = seconds;
        _onEarth = _seconds / 31557600;
    }

    double getSeconds() {
        return _seconds;
    }

    double onEarth() {
        return _onEarth;
    }

    double onMercury() {
        return _onEarth * 0.2408467;
    }

    double onVenus() {
        return _onEarth * 0.61519726;
    }

    double onMars() {
        return _onEarth * 1.8808158;
    }

    double onJupiter() {
        return _onEarth * 11.862615;
    }

    double onSaturn() {
        return _onEarth * 29.447498;
    }

    double onUranus() {
        return _onEarth * 84.016846;
    }

    double onNeptune() {
        return _onEarth * 164.79132;
    }
}

File: csharp/space-age/SpaceAge.cs
using System;

public class SpaceAge
{
    private static double earthYears;
    public SpaceAge(int seconds)
    {
        earthYears = seconds / 31557600.00;
    }

    public double OnEarth()
    {
        return earthYears;
    }

    public double OnMercury()
    {
        return earthYears / 0.2408467;
    }

    public double OnVenus()
    {
        return earthYears / 0.61519726;
    }

    public double OnMars()
    {
        return earthYears / 1.8808158;
    }

    public double OnJupiter()
    {
        return earthYears / 11.862615;
    }

    public double OnSaturn()
    {
        return earthYears / 29.447498;
    }

    public double OnUranus()
    {
        return earthYears / 84.016846;
    }

    public double OnNeptune()
    {
        return earthYears / 164.79132;
    }
}